src/app/main/service/habit-assign/habit-assign.service.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { LanguageService } from './../../i18n/language.service';
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { HabitAssignService } from './habit-assign.service';
+import { habitAssignLink } from '../../links';
+import {
+  listOfHabits,
+  newHabit,
+  assignResponce,
+  modifiedAssignResponce,
+  HabitsForDate,
+  habitsWithTheSameHabitId
+} from '../../mocks/habit-assign-mock';
+
+describe('HabitService', () => {
+  let service: HabitAssignService;
+  let httpMock: HttpTestingController;
+  const languageServiceMock = jasmine.createSpyObj('LanguageService', ['getCurrentLanguage']);
+  languageServiceMock.getCurrentLanguage = () => 'en';
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [HabitAssignService, { provide: LanguageService, useValue: languageServiceMock }],
+      imports: [HttpClientTestingModule]
+    }),
+      (httpMock = TestBed.get(HttpTestingController));
+    service = TestBed.get(HabitAssignService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+  it('should get data', () => {
+    service.getAssignedHabits().subscribe((data) => {
+      expect(data).toEqual(listOfHabits);
+    });
```

Comment:
Look, in such methods where we are just doing some HTTP calls without any additional logic inside we can just check if the `http.(get/post/patch)` is doing the right requests with the right params
It's redundant to subscribe to your mocked methods as you will not check anything
You will know only that `req.flush()` method is working well =)
So it would be enough just to spy on `http.(get/post/patch)` method and check it with `.toHaveBeenCalledWith`